### PR TITLE
[stable/traefik] Add acme dns resolvers configuration

### PR DESCRIPTION
**Pull request by emmanuelCarre**
#### What this PR does / why we need it:

In some case, we need to define DNS servers for Letsencrypt DNS challange with option `acme.dnsChallenge.resolvers` (see [acme configuration](https://docs.traefik.io/configuration/acme/)) but is not available into Helm deployment.

#### Which issue this PR fixes
None

#### Special notes for your reviewer:
* I was not sure if I have to squash my commit...
* Naming variables is never easy; Don't hesitate to suggest better names.

#### Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/traefik/README.md [hunk 1]
```diff
@@ -140,6 +140,8 @@ The following table lists the configurable parameters of the Traefik chart and t
 | `acme.domains.domainsList`             | List of domains & (optional) subject names                                                                                   | `[]`                                              |
 | `acme.domains.domainsList.main`        | Main domain name of the generated certificate                                                                                | *.example.com                                     |
 | `acme.domains.domainsList.sans`        | optional list of alternative subject names to give to the certificate                                                        | `[]`                                              |
+| `acme.resolvers.enable`                | Whether to enable specific DNS servers for DNS challenge                                                                     | `false`                                           |
```
Comment (dtomcej) [inline stable/traefik/README.md]: Does this actually do anything?

File: stable/traefik/templates/configmap.yaml [hunk 2]
```diff
@@ -252,6 +252,9 @@ data:
     {{- if eq .Values.acme.challengeType "dns-01" }}
       [acme.dnsChallenge]
       provider = {{ .Values.acme.dnsProvider.name | quote }}
+      {{- if .Values.acme.resolvers.enable }}
```
Comment (dtomcej) [inline stable/traefik/templates/configmap.yaml]: Can you just do `if .Values.acme.resolvers` and skip the whole enable value?

File: stable/traefik/values.yaml [hunk 3]
```diff
@@ -174,6 +174,11 @@ acme:
   ## Configure dnsProvider to perform domain verification using dns challenge
   ## Applicable only if using the dns-01 challenge type
   delayBeforeCheck: 0
+  resolvers:
```
Comment (dtomcej) [inline stable/traefik/values.yaml]: That would make this:

```yaml
resolvers: []
# - 1.1.1.1:53
# - 8.8.8.8:53
```

**Review by dtomcej — CHANGES_REQUESTED** [review]

File: stable/traefik/templates/configmap.yaml (hunk 2 above)
Comment (emmanuelCarre) [reply stable/traefik/templates/configmap.yaml]: Done. I was afraid about empty array is always `true` but it is work fine.  Thx for this tips :)

File: stable/traefik/README.md (hunk 1 above)
Comment (emmanuelCarre) [reply stable/traefik/README.md]: It is not needed with with `acme.resolvers: []` in `values.yaml`. I removed this change.

File: stable/traefik/values.yaml (hunk 3 above)
Comment (emmanuelCarre) [reply stable/traefik/values.yaml]: Done.

**Comment by emmanuelCarre** [comment]
Newbee contributor question: Is it fine if I solve directly conversation after take care of feedback?

**Comment by ldez** [comment]
@emmanuelCarre could you rebase ?

**Comment by emmanuelCarre** [comment]
Hi,
I rebase, it should be OK now.

**Comment by ldez** [comment]
@emmanuelCarre could you rebase once again ?

**Comment by emmanuelCarre** [comment]
Done :)

**Pull request merged**